flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlChunkSplitter.java

```diff
@@ -309,14 +309,17 @@ private Object nextChunkEnd(
             int chunkSize)
             throws SQLException {
         // chunk end might be null when max values are removed
-        Object chunkEnd =
-                queryNextChunkMax(jdbc, tableId, splitColumnName, chunkSize, previousChunkEnd);
+        Object[] chunkEndAndCount =
+                queryNextChunkMaxAndCount(
+                        jdbc, tableId, splitColumnName, chunkSize, previousChunkEnd);
+        Object chunkEnd = chunkEndAndCount[0];
+        Integer chunkCount = Integer.valueOf(chunkEndAndCount[1].toString());
```

Comment:
The `chunkCount` will never be larger than `chunkSize` as `chunkCount` is selected by `LIMIT chunkSize` sql. Then `chunkSize > chunkCount` will always be false.